### PR TITLE
Bamboo Overhaul (Port of Beestation Hornet #5703)

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
Look
Pea green bamboo just... doesn't do it.
image
THIS? This is Bamboo.

I've been working with the spriter of the bamboo in https://github.com/BeeStation/BeeStation-Hornet/pull/5703, who was actually wanting to port it up to here for a while now.
All existing icons have been replaced, the material datum's color updated, and several new items created; also, the plant has 1 more growth stage (graphically) to make it smoother - this doesnt affect speed, just another icon.

Included ADDITIONS:
- Bamboo spear, which is slightly better for throwing
- Crude Syringes, with only 5u max capacity, but disposable
- Walls/Falsewalls and Floors
- Benches and Stools

Included CHANGES:
- New Blowpipe icons
- New stack icons
- New growth icons
- New Punji sprites
- New datum color (for carving block)
- Head-Spear variable changes : **REVIEW, AS IDK WHY BUT IT DOESNT WANNA WORK, AND ALL THE USED ITEMS CAN BE INTERACTED WITH**

**All sprite credit goes to Pirill**

## Why It's Good For The Game
Better sprites for an underused gem. Maybe now the glory of Bamboo will be realized.

## Changelog

:cl: Orion_the_Fox, Pirill
add: Added several new Bamboo items, turfs, and more!
imageadd: Updated existing Bamboo items to a less repulsive green.
/:cl:

### CODE REVIEW

**Comment by OrionTheFox** [comment]
Now that I've posted this, time to go to sleep 😎 
Working on it all night was a mistake

**Comment by Ebb-Real** [comment]
spessman so skilled they can make syringes out of bamboo with their bare hands. Love it.

**Comment by TheBonded** [comment]
Considering they can take a bone saw, some sheets of plasteel, and a handful of electrical wires and weld them together into a fully functioning chainsaw, making a syringe out of bamboo is probably fine.

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 1]
```diff
@@ -124,6 +124,15 @@
...
 	merge_type = /obj/item/stack/tile/wood/tile
 
+//Bamboo
+/obj/item/stack/tile/bamboo
+	name = "bamboo mat pieces"
+	singular_name = "bamboo mat piece"
+	desc = "A piece of a bamboo mat with a decorative trim."
+	icon_state = "tile_bamboo"
+	turf_type = /turf/open/floor/bamboo
+	resistance_flags = FLAMMABLE
```
Comment (OrionTheFox) [inline code/game/objects/items/stacks/tiles/tile_types.dm:134]:
```suggestion
/obj/item/stack/tile/bamboo
	name = "bamboo mat pieces"
	singular_name = "bamboo mat piece"
	desc = "A piece of a bamboo mat with a decorative trim."
	icon_state = "tile_bamboo"
	turf_type = /turf/open/floor/bamboo
	merge_type = /obj/item/stack/tile/bamboo
	resistance_flags = FLAMMABLE
	
```

**Comment by Archanial** [comment]
Remember to include https://github.com/BeeStation/BeeStation-Hornet/pull/5895

**Comment by OrionTheFox** [comment]
> Remember to include [BeeStation/BeeStation-Hornet#5895](https://github.com/BeeStation/BeeStation-Hornet/pull/5895)

already done 👍

**Comment by OrionTheFox** [comment]
``` 
## TESTING: GC: -- [0x20188af] | /obj/item/toy/cards/deck/syndicate was unable to be GC'd --
FAIL: /datum/unit_test/create_and_destroy 724.7s
	REASON #1: /obj/item/toy/cards/deck/syndicate hard deleted 1 times out of a total del count of 1
```
CI fail unrelated to my own code

**Review by Krysonism — APPROVED** [review]

**Review by MrMelbert — APPROVED** [review]
Resolve conflicts please

**Pull request merged**